allennlp/data/token_indexers/single_id_token_indexer.py

```diff
@@ -43,7 +55,7 @@ def tokens_to_indices(self,
                           index_name: str) -> Dict[str, List[int]]:
         indices: List[int] = []
 
-        for token in tokens:
+        for token in itertools.chain(self._start_tokens, tokens, self._end_tokens):
```

Comment:
I need to remember to use itertools more often...  I would have just done `self._start_tokens + tokens + self._end_tokens`, which is less efficient.